effectsize_example.Rmd

```diff
@@ -0,0 +1,84 @@
+---
+title: "Effect size"
+output: html_notebook
+---
+```{r}
+library(tidyverse)
+```
+
+
+## Data
+Assuming between-subjects design:
+```{r}
+set.seed(12)
+n <- 20
+data <- tibble(
+  Group = rep(c("A", "B"), each = n),
+  TaskCompletionTime_Sec = c(
+    rnorm(n, mean = 12, sd = 1),
+    rnorm(n, mean = 10, sd = 2)
+  )
+)
+```
+
+## Visualizing data
+
+```{r, fig.height=2, fig.width=3}
+pd <- position_dodge(0.2)
+data %>% 
+  ggplot(aes(x = Group, y = TaskCompletionTime_Sec)) +
+  geom_point(alpha = 0.2, position = pd) +
+  stat_summary(fun.data = "mean_cl_normal", geom = "pointrange") +
+  coord_flip() +
+  ylab("Task Completion Time (s)")
+
+
+```
+
+## Simple effect size
+
+```{r}
+data_A <- (data %>% filter(Group == "A"))[[2]]
+data_B <- (data %>% filter(Group == "B"))[[2]]
+t_result <- tidy(t.test(data_A, data_B))
+```
+
+*Simple effect size:* The task completion time of the two groups differ by `r t_result$estimate` (95% confidence interval [`r t_result$conf.low`, `r t_result$conf.high`]) seconds.
+
+```{r, fig.height=1, fig.width=3}
+library(broom)
+t_result %>% 
+  ggplot(aes(x = 1, y = estimate, ymin = conf.low, ymax = conf.high)) +
+  geom_pointrange() +
+  geom_hline(yintercept = 0, color = "red") +
+  coord_flip() +
+  ylab("Difference in task completion time (s)") +
+  theme(axis.title.y = element_blank(), axis.text.y = element_blank(), axis.ticks.y = element_blank())
+```
+
+## Standardized effect size
+
+```{r}
+library(effsize)
+cohen_d <- cohen.d(TaskCompletionTime_Sec ~ Group, data = data)
```

Comment:
Is it worth showing the manual cohen's d calculation as well, to demystify it a bit? E.g. show the calculation then say "happily, there's a library that does this for us as well".